### PR TITLE
fix focusable custom-element tabbing issue

### LINKED ISSUE #1068: Bug with selecting jumping to first tabbable from last if web component is involved.
## Description

I found an issue where if you put a focus trap on a web component, like you have a `my-modal` element, and also add `tabindex="0"` to it to make it focusable/tabbable, when you cycle through the tabbable elements within the web component with `Tab`, you cannot jump back to the first element from the last.

## Expected behaviour

When I hit `Tab` on the last element, the containing web-component, which is the first tabbable element is focused

## Actual behaviour

Hitting `Tab` on the last element seemingly does nothing

## Reproduction

More detailed steps on how you can see the issue are provided in this stackblitz

https://stackblitz.com/edit/vitejs-vite-mxdsxb?file=index.html

## Suggestion

I identified that the issue comes from here: https://github.com/focus-trap/focus-trap/blob/ed95b7caf463170f972e876bdb58c46c9264a122/index.js#L423-L425

The `node` that we want to focus is going to be the web-component itself, and `document.activeElement` will also be the web-component, but that's not really true, because inside its shadow-root, there's another activeElement, but that's not visible from the outside this way.

My proposed solution is that when doing this check, it should recursively check if `document.activeElement` has a shadow-root and if it does, check for the `activeElement` inside that.

If someone from the project can confirm that this is a bug and my solution seems okay, I'm happy to submit a PR. If you think another solution would work better, I'm happy to work on that as well.

### PR DESCRIPTION
**Pull request by simonxabris**
fixes #1068 

Added a function that recursively checks for the `activeElement` and if the `activeElement` is a custom-element, it pierces the shadow-dom and checks for `activeElement` in it to determine which element is truly in focus.

I was not sure how to properly test this as imo it's very much an edge case, so for now I just updated the current `in-open-shadow-dom` example to have the structure I described in the issue and manually checked if my changes fix that. I have not written any extra tests for now, but if you think that new tests are warranted, I'm happy to add them

aside: I also added an `all-contributos` script to `package.json`, as according to to [cotribution docs](https://github.com/focus-trap/focus-trap/blob/master/CONTRIBUTING.md#anything-helps) it already should've been there.

<details>
<summary>PR Checklist</summary>
<br/>

__Please leave this checklist in your PR.__

- Issue being fixed is referenced.
- Source changes maintain stated browser compatibility.
- Web APIs introduced have __deep__ browser coverage, including Safari (often very late to adopt new APIs).
- Includes updated docs demo bundle if source/docs code was changed (run `npm run demo-bundle` in your branch and include the `/docs/demo-bundle.js` file that gets generated in your PR).
- Unit test coverage added/updated.
- E2E (i.e. demos) test coverage added/updated.
  - ⚠️ Non-covered demos (look for `// TEST MANUALLY` comments [here](https://github.com/focus-trap/focus-trap/blob/master/docs/js/index.js)) that can't be fully tested in Cypress have been __manually__ verified.
- Typings added/updated.
- Changes do not break SSR:
  - Careful to test `typeof document/window !== 'undefined'` before using it in code that gets executed on load.
- README updated (API changes, instructions, etc.).
- Changes to dependencies explained.
- Changeset added (run `npm run changeset` locally to add one, and follow the prompts).
  - EXCEPTION: A Changeset is not required if the change does not affect any of the source files that produce the package bundle. For example, demo changes, tooling changes, test updates, or a new dev-only dependency to run tests more efficiently should not have a Changeset since it will not affect package consumers.

</details>

### CODE REVIEW

File: cypress/downloads/downloads.html
[the remark this answers is not in the document]
Comment (simonxabris) [reply cypress/downloads/downloads.html:1]: This is what I could find about this in the cypress docs: https://docs.cypress.io/guides/references/configuration#Downloads

Not sure why this was generated for me, I'll remove it and add it to `.gitignore` as you recommended.

**Comment by simonxabris** [comment]
Fixed everything you pointed out in the review, this is how the focus states look now:

<details>
<summary>Focus demo</summary>


https://github.com/focus-trap/focus-trap/assets/27497229/d813b320-384a-45b3-bc0f-b7fd23c66852



</details>

**Comment by stefcameron** [comment]
> Fixed everything you pointed out in the review, this is how the focus states look now:
> 
> Focus demo
> I added a focus ring around the whole element, as it's now focusable because of the `tabindex="0"` is added to the custom element

Nice, thank you! The demo looks great now in your video. I'll check out your changes ASAP!

**Review by stefcameron — APPROVED** [review]
Nice! Thank you for this PR. Everything looks good to me. 😄

**Pull request merged**